lib/zanzibar.rb

```diff
@@ -1,210 +1,167 @@
-          file.puts Base64.decode64(response[:envelope][:body][:download_file_attachment_by_item_id_response][:download_file_attachment_by_item_id_result][:file_attachment])
-        end
-      rescue Savon::Error => err
-        raise "There was an error getting the attachment from secret #{args[:scrt_id]}: #{err}"
-      end
-    end
-  end
-end
+require "zanzibar/version"
+require 'savon'
+require 'io/console'
+require 'fileutils'
+
+module Zanzibar
+
+  ##
+  # Class for interacting with Secret Server
+  class Zanzibar
+
+    ##
+    # @param args{:domain, :wsdl, :pwd, :username, :globals{}}
+
+    def initialize(args = {})
+
+      if args[:username]
+        @@username = args[:username]
+      else
+        @@username = ENV['USER']
+      end
+
+      if args[:wsdl]
+        @@wsdl = args[:wsdl]
+      else
+        @@wsdl = get_wsdl_location
+      end
+      if args[:pwd]
+          @@password = args[:pwd]
+      else
+        @@password = prompt_for_password
+      end
+      if args[:domain]
+        @@domain = args[:domain]
+      else
+        @@domain = prompt_for_domain
+      end
+      args[:globals] = {} unless args[:globals]
+      init_client(args[:globals])
+    end
+
+    ## Initializes the Savon client class variable with the wdsl document location and optional global variables
+    # @param globals{}, optional
+
+    def init_client(globals = {})
+      globals = {} if globals == nil
+      @@client = Savon.client(globals) do
+        wsdl @@wsdl
+      end
+    end
+
+    ## Gets the user's password if none is provided in the constructor.
+    # @return [String] the password for the current user
+
+    def prompt_for_password
+      puts "Please enter password for #{@@username}:"
+      return STDIN.noecho(&:gets).chomp
+    end
+
+    ## Gets the wsdl document location if none is provided in the constructor
+    # @return [String] the location of the WDSL document
+
+    def prompt_for_wsdl_location
+      puts "Enter the URL of the Secret Server WSDL:"
+      return STDIN.gets.chomp
+    end
+
+    ## Gets the domain of the Secret Server installation if none is provided in the constructor
+    # @return [String] the domain of the secret server installation
+
+    def prompt_for_domain
+      puts "Enter the domain of your Secret Server:"
+      return STDIN.gets.chomp
+    end
+
+
+    ## Get an authentication token for interacting with Secret Server. These are only good for about 10 minutes so just get a new one each time.
+    # Will raise an error if there is an issue with the authentication.
+    # @return the authentication token for the current user.
+
+    def get_token
+      begin
+        response = @@client.call(:authenticate, message: { username: @@username, password: @@password, organization: "", domain: @@domain }).hash[:envelope][:body][:authenticate_response][:authenticate_result]

```

Comment:
nit - perhaps a `get_authenticate_result` function that does this hashwalking somewhere else to shorten the line and make it more clear what is happening?